quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/TruststoreBuilder.java

```diff
@@ -0,0 +1,141 @@
+package org.keycloak.quarkus.runtime;
+
+import org.jboss.logging.Logger;
+import org.keycloak.config.SecurityOptions;
+import org.keycloak.quarkus.runtime.configuration.Configuration;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+
+/**
+ * Builds a system-wide truststore from the given config options.
+ */
+public class TruststoreBuilder {
+
+    private static final String SYSTEM_TRUSTSTORE_KEY = "javax.net.ssl.trustStore";
+    private static final String PKCS12 = "PKCS12";
+
+    private static final Logger LOGGER = Logger.getLogger(TruststoreBuilder.class);
+
+    public static void setSystemTruststore() {
+        String[] truststores = Configuration.getOptionalKcValue(SecurityOptions.TRUSTSTORE.getKey()).map(s -> s.split(",")).orElse(null);
+
+        if (truststores == null || truststores.length == 0) {
+            return;
+        }
+
+        boolean trustStoreIncludeDefault = Configuration.getOptionalBooleanKcValue(SecurityOptions.TRUSTSTORE_INCLUDE_DEFAULT.getKey()).orElse(false);
+
+        KeyStore truststore = createPkcs12KeyStore();
+
+        if (trustStoreIncludeDefault) {
+            File defaultTrustStore = getDefaultTrustStoreFile();
+            if (defaultTrustStore.exists()) {
+                String trustStoreType = System.getProperty("javax.net.ssl.trustStoreType", KeyStore.getDefaultType());
+                String path = defaultTrustStore.getAbsolutePath();
+                mergeTrustStore(truststore, path, loadStore(path, trustStoreType, null));
+            } else {
+                LOGGER.warnf("Default truststore was to be included, but could not be found at: %s", defaultTrustStore);
+            }
+        }
+
+        for (String file : truststores) {
+            if (file.endsWith(".p12") || file.endsWith(".pfx")) {
+                mergeTrustStore(truststore, file, loadStore(file, PKCS12, null));
+            } else {
+                mergePemFile(truststore, file);
+            }
+        }
+
+        String confDir = System.getProperty("jboss.server.config.dir");
```

Comment:
We should not use jboss properties as they'll likely go away. We could probably use something like:
https://github.com/keycloak/keycloak/blob/e3af0610e21961c1e8d074a223f9fffee0e56dde/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/CachingPropertyMappers.java#L45-L51

But we could possibly use the `data` dir (it's used e.g. by h2).

Reply:
Switched to the data dir from the Environment utility.